app/models/UserAnswers.scala

```diff
@@ -63,6 +65,9 @@ final case class UserAnswers(
         page.cleanup(None, updatedAnswers)
     }
   }
+
+  def schemeBankAccounts(srn: Srn): List[BankAccount] =
```

Comment:
This class should not really contain any logic to any specific page

One thing you could do is have an object Page as well as a case class Page that onlt implements the `Getter[A]` and returns the whole collection rather than a single element. Somthing like
```
case class SchemeBankDetailsPage(srn: Srn, index: Int) extends QuestionPage[BankAccount] {
    def path: JsPath = JsPath \ toString \ index
    def toString: String = SchemeBankDetailsPage.pageName
}

object SchemeBankDetailsPage extends Gettable[List[BankAccount]] {
    val pageName: String = ???
    def path: JsPath = JsPath \ pageName 
}
```